File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/location/replay/ReplayRouteLocationListener.java
package com.mapbox.services.android.navigation.v5.location.replay;

import android.location.Location;

import com.mapbox.android.core.location.LocationEngineCallback;
import com.mapbox.android.core.location.LocationEngineResult;

class ReplayRouteLocationListener implements ReplayLocationListener {

  private final ReplayRouteLocationEngine engine;
  private final LocationEngineCallback<LocationEngineResult> callback;

  ReplayRouteLocationListener(ReplayRouteLocationEngine engine,
                              LocationEngineCallback<LocationEngineResult> callback) {
    this.engine = engine;
    this.callback = callback;
  }

  @Override
  public void onLocationReplay(Location location) {
    engine.updateLastLocation(location);
    engine.removeLastMockedLocation();
    LocationEngineResult result = LocationEngineResult.create(location);
    callback.onSuccess(result);
  }
}

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/network/replay/ReplayRouteLocationListener.kt
package com.mapbox.navigation.base.network.replay

import android.location.Location
import com.mapbox.android.core.location.LocationEngineCallback
import com.mapbox.android.core.location.LocationEngineResult
import com.mapbox.navigation.base.network.ReplayRouteLocationEngine

internal class ReplayRouteLocationListener(
    private val engine: ReplayRouteLocationEngine,
    private val callback: LocationEngineCallback<LocationEngineResult>
) : ReplayLocationListener {

    override fun onLocationReplay(location: Location) {
        engine.updateLastLocation(location)
        engine.removeLastMockedLocation()
        val result = LocationEngineResult.create(location)
        callback.onSuccess(result)
    }
}
